### PR TITLE
[Test] Restart the database using three modes

### PR DESCRIPTION
**Pull request by JingchenZhang**

### CODE REVIEW

File: db_test.go [hunk 1]
```diff
@@ -532,10 +532,73 @@ func withRAMIdxDB(t *testing.T, fn func(t *testing.T, db *DB)) {
...
+
+func Test_Three_EntryIdexMode_RestartDB(t *testing.T) {
+
+	tmpdir, _ := os.MkdirTemp("", "nutsdb")
+	opt := DefaultOptions
+	opt.Dir = tmpdir
+	opt.EntryIdxMode = HintKeyValAndRAMIdxMode
+	opt.SegmentSize = 8 * 1024
+
+	withDBOption(t, opt, func(t *testing.T, db *DB) {
```
Comment (elliotchenzichang) [inline db_test.go:551]: I can recall that @bigboss2063 refacted the code which runs test case with customized database. Could you help also refactor this part of code when you have time? For more context and code style you can refer this artcle.  Thanks！https://mp.weixin.qq.com/s/w8YaLU0b3bEzUXdrxezodQ

File: merge_test.go [hunk 2]
```diff
@@ -15,11 +15,12 @@
 package nutsdb
 
 import (
-	"github.com/stretchr/testify/require"
-	"github.com/xujiajun/utils/strconv2"
 	"sync"
 	"testing"
 	"time"
+
+	"github.com/stretchr/testify/require"
```
Comment (elliotchenzichang) [inline merge_test.go:22]: Why only added dependency in this file? Is that expected?

**Review by elliotchenzichang — COMMENTED** [review]
Left some comments!

File: db_test.go (hunk 1 above)
Comment (elliotchenzichang) [reply db_test.go:551]: And this is the related PR link!  https://github.com/nutsdb/nutsdb/pull/375

Comment (JingchenZhang) [reply db_test.go:551]: ok

File: merge_test.go (hunk 2 above)
Comment (JingchenZhang) [reply merge_test.go]: I don't know, maybe my editor did it, I'll check it out

**Review by elliotchenzichang — COMMENTED** [review]
LGTM！Thanks for this great contribution!

**Pull request merged**